### ISSUE TITLE
Robot Briefing Issue: D2 Vertigo high-resolution robot movie RB9 fails to play with PhysFS 3

### ISSUE THREAD

**Issue by LordKane1701** [dxx-rebirth/dxx-rebirth#427]
### Environment
https://www.dxx-rebirth.com/download/dxx/rebirth/dxx-rebirth_20190530-win.zip

#### Operating System Environment

* Microsoft Windows
  * [ ] Microsoft Windows XP
  * [ ] Microsoft Windows Vista
  * [ X] Microsoft Windows 7
  * [ ] Microsoft Windows 8
  * [ ] Microsoft Windows 8.1
  * [ ] Microsoft Windows 10

* Mac OS X
  * [ ] Mac OS X 10.8
  * [ ] Mac OS X 10.9
  * [ ] Mac OS X 10.10
  * [ ] Mac OS X 10.11
  * [ ] Mac OS X 10.12
  * [ ] Mac OS X 10.13

* Debian
  * [ ] Debian Wheezy
  * [ ] Debian Jessie
  * [ ] Debian Stretch
  * [ ] Debian Buster
* Fedora
  * [ ] Fedora 26
  * [ ] Fedora 27
  * [ ] Fedora 28
  * [ ] Fedora 29
  * [ ] Fedora 30
  * [ ] Rawhide
* Ubuntu
  * [ ] Ubuntu 16.04 LTS (Xenial Xerus)
  * [ ] Ubuntu 16.10 (Yakkety Yak)
  * [ ] Ubuntu 17.04 (Zesty Zapus)
  * [ ] Ubuntu 17.10 (Artful Aardvark)
  * [ ] Ubuntu 18.04 LTS (Bionic Beaver)
  * [ ] Ubuntu 18.10 (Cosmic Cuttlefish)
  * [ ] Ubuntu 19.04 (Disco Dingo)

* [ ] Arch
* [ ] Gentoo

* [ ] OpenBSD
* [ ] FreeBSD
* [ ] NetBSD

#### CPU environment

* [ ] x86 (32-bit Intel/AMD)
* [ X] x86\_64 (64-bit Intel/AMD)
* [ ] ARM (32-bit)
* [ ] ARM64 (64-bit; sometimes called AArch64)

#### Game environment
Descent 2: Vertigo

### Description
When you start Descent 2 Vertigo, and the introductry cutscene plays, the movie for the Fervid 99 is not displayed. 

#### Regression status
As far as I know this did not occur in earlier builds. 

### Steps to Reproduce
Install Descent 2: Vertigo, Start campaign watch Briefing movies 

* Engines affected:
  * [ ] D1X-Rebirth
  * [ X] D2X-Rebirth

* Game modes affected:
  * [ X] Single player
  * [ ] Multiplayer competitive
    * [ ] Anarchy
    * [ ] Team Anarchy
    * [ ] Robo-Anarchy
    * [ ] Capture the Flag
    * [ ] Bounty
  * [ ] Multiplayer cooperative

* Players affected by issue:
  * [X ] Game host
  * [ ] Game guests

Screenshot and Gamelog provided below, if you require more info just ask, I will do my best to provide what I can. 

[gamelog.txt](https://github.com/dxx-rebirth/dxx-rebirth/files/3261355/gamelog.txt)
![2019-06-06 11-14-24](https://user-images.githubusercontent.com/26239277/59029647-0bcda880-881c-11e9-8269-e1e2c0d739ab.png)

**Comment by zicodxx** [comment]
@LordKane1701 The robots in Descent 2 and the Vertige Addon are both stored in movie files. For Vertigo that would be d2x-h.mvl (for the hires movies) and d2x-l.mvl (for the lowres movies). The game can technically work without those movies, but then you get these empty frames. Are those two files present in your installation?

**Comment by LordKane1701** [comment]
> 
> 
> @LordKane1701 The robots in Descent 2 and the Vertige Addon are both stored in movie files. For Vertigo that would be d2x-h.mvl (for the hires movies) and d2x-l.mvl (for the lowres movies). The game can technically work without those movies, but then you get these empty frames. Are those two files present in your installation?

Yes I can confirm that both files are present in my installation.

**Comment by vLKp** [comment]
Do the same movies work for you when you use an older snapshot of Rebirth?  In what directory did you place the Vertigo movie files (`d2x-h.mvl`)?  The correct location is the top level `.d2x-rebirth` directory.

**Comment by LordKane1701** [comment]
> 
> 
> Do the same movies work for you when you use an older snapshot of Rebirth? In what directory did you place the Vertigo movie files (`d2x-h.mvl`)? The correct location is the top level `.d2x-rebirth` directory.

Yes they worked on a eariler build I sadly do not recall which, it was one after Beta 2 and yes they are in the top level directory, also I forgot to mention it is as far as I know only the fervid 99 or what is affected, I have not seen later robot briefings affected, but the one before that one (at the start of a new campaign) works fine, its just as far as I know the its just the fervid 99 briefing. I'll do a complete map run through at a later time to see if others are affect and post updated screenshots and gamelogs.

**Comment by LordKane1701** [comment]
Alright after checking all 20 levels of vertigo. only and ONLY the fervid 99 briefing is affected. I have tried 3 different vertigo discs (that I own) 4 "pirate" downloads of the vertigo data and one copy from an online friend and all of them result in the same thing Fervid 99 robot does not display as pictured.

**Comment by zicodxx** [comment]
I bisected this and while I didn't find a revision that had all movies work, I found the revision where all movies were broken and then subsequently fixed in: f2f41cd60e7d57a5188f5ea7a35c0c11512bbe61

The commit message links to a discussion about the bug, where a comment points out the issue we have here:
https://github.com/dxx-rebirth/dxx-rebirth/issues/379#issuecomment-390519276

The issue further mentions that PhysFS 3.0.2 should fix that, but while I run 3.0.2, I can still observe the issue.

**Comment by LordKane1701** [comment]
is it possible to open the MVL file and fix the captailization directly?

**Comment by vLKp** [comment]
[The last comment (as of this writing)](https://github.com/dxx-rebirth/dxx-rebirth/issues/379#issuecomment-477790175) says that PhysFS 3.0.2 doesn't actually contain the required fix.

> is it possible to open the MVL file and fix the captailization directly?

Yes.  The file is not integrity checked.  You may edit it freely.

**Comment by LordKane1701** [comment]
> 
> 
> [The last comment (as of this writing)](https://github.com/dxx-rebirth/dxx-rebirth/issues/379#issuecomment-477790175) says that PhysFS 3.0.2 doesn't actually contain the required fix.
> 
> > is it possible to open the MVL file and fix the captailization directly?
> 
> Yes. The file is not integrity checked. You may edit it freely.

Any idea how to open a MVL file?

**Comment by zicodxx** [comment]
@vLKp That's correct. Since the issue is still relevant as of now, I can slap together a small helper function that converts the two filenames in question (or potentially use ignorecase, although I am not sure if that would fail due to the issue) which we could use until the problem is formally addressed. Alternatively, I can compile the next Windows build using an older version of PhysFS, but that doesn't necessarily help players that are on other platforms.

**Comment by vLKp** [comment]
> Any idea how to open a MVL file?

Any hex editor should do, since you don't need to change the file's length.  I wouldn't seriously suggest changing this though.  There's very little value to fixing this movie.

-----

Now that I have PhysFS 3.0 installed here, I see the problem on Linux, so we should at least release-note the issue if we don't add an in-code workaround.

I'm not inclined to try to work around it in code, since `InitRobotMovie` needs to adjust the case, but `init_movie(const char *movielib, int required, loaded_movie_t &movie)` is the only site which knows whether the game picked the high or the low resolution movie, and we need to know that in order to decide how to adjust the filename.

To recap the findings posted by @ryusei117 in the prior issue:

D2X-H.MVL | D2X-L.MVL
--------- | ---------
RB9.mve | rb9.mve
RB$.MVE | RB$.mve

File names _should_ always be `RBx.MVE`, where `x` is set by the briefing and all other characters are literal.  In `D2X-H.MVL`, only `RB9` is wrong (lowercase extension).  In `D2X-L.MVL`, `RB9` is doubly wrong (lowercase everything) and `RB$` is wrong (lowercase extension).

**Comment by LordKane1701** [comment]
> 
> 
> > Any idea how to open a MVL file?
> 
> Any hex editor should do, since you don't need to change the file's length. I wouldn't seriously suggest changing this though. There's very little value to fixing this movie.
> 
> Now that I have PhysFS 3.0 installed here, I see the problem on Linux, so we should at least release-note the issue if we don't add an in-code workaround.
> 
> I'm not inclined to try to work around it in code, since `InitRobotMovie` needs to adjust the case, but `init_movie(const char *movielib, int required, loaded_movie_t &movie)` is the only site which knows whether the game picked the high or the low resolution movie, and we need to know that in order to decide how to adjust the filename.
> 
> To recap the findings posted by @ryusei117 in the prior issue:
> D2X-H.MVL 	D2X-L.MVL
> RB9.mve 	rb9.mve
> RB$.MVE 	RB$.mve
> 
> File names _should_ always be `RBx.MVE`, where `x` is set by the briefing and all other characters are literal. In `D2X-H.MVL`, only `RB9` is wrong (lowercase extension). In `D2X-L.MVL`, `RB9` is doubly wrong (lowercase everything) and `RB$` is wrong (lowercase extension).

Yes there is little value, but I need it to appease my sense of completeness, the hex editing worked, I am going to consider this closed in the mean time, thank you for all your assistance vLKp and Zico :) I hope you get this resolved in the future.  I can also write up a hex editing guide if people need it.

**Comment by LordKane1701** [comment]
Ooops forgot to close. if anyone needs help doing hex editing on D2X-H.MVL to "fix" this issue, hit me up on the DXX rebirth official discord.

**Comment by vLKp** [comment]
If the instructions are simple, you could post them here so that readers do not need to find you later.  I expect it's just changing the bytes at particular offsets, correct?  Which offsets, and to what values should they be set?

**Comment by LordKane1701** [comment]
I just used the hex edit plugin on notepad ++ and changed the
capitalization to what you stated on the chart above.
